code/datums/gamemode/factions/legacy_cult/runes.dm

```diff
@@ -0,0 +1,1621 @@
+								//ticker.mode.update_cult_icons_added(C.mind)
+					else
+						var/list/construct_types = list("Artificer", "Wraith", "Juggernaut")
+						construct_class = input("Please choose which type of construct you wish [M] to become.", "Construct Transformation") in construct_types
+						switch(construct_class)
+							if("Juggernaut")
+								var/mob/living/simple_animal/construct/armoured/C = new /mob/living/simple_animal/construct/armoured (get_turf(src.loc))
+								M.mind.transfer_to(C)
+								qdel(M)
+								M = null
+								to_chat(C, "<B>You are now a Juggernaut. Though slow, your shell can withstand extreme punishment, create temporary walls and even deflect energy weapons, and rip apart enemies and walls alike.</B>")
+								//ticker.mode.update_cult_icons_added(C.mind)
+							if("Wraith")
+								var/mob/living/simple_animal/construct/wraith/C = new /mob/living/simple_animal/construct/wraith (get_turf(src.loc))
+								M.mind.transfer_to(C)
+								qdel(M)
+								M = null
+								to_chat(C, "<B>You are a now Wraith. Though relatively fragile, you are fast, deadly, and even able to phase through walls.</B>")
+								///ticker.mode.update_cult_icons_added(C.mind)
+							if("Artificer")
+								var/mob/living/simple_animal/construct/builder/C = new /mob/living/simple_animal/construct/builder (get_turf(src.loc))
+								M.mind.transfer_to(C)
+								qdel(M)
+								M = null
+								to_chat(C, "<B>You are now an Artificer. You are incredibly weak and fragile, but you are able to construct new floors and walls, to break some walls apart, to repair allied constructs (by clicking on them), </B><I>and most important of all create new constructs</I><B> (Use your Artificer spell to summon a new construct shell and Summon Soulstone to create a new soulstone).</B>")
+								//ticker.mode.update_cult_icons_added(C.mind)
+								for(var/spell/S in C.spell_list)
+									if(S.charge_type & Sp_RECHARGE)
+										if(S.charge_counter == S.charge_max) //Spell is fully charged - let the proc handle everything
+											S.take_charge()
+										else //Spell is on cooldown and already recharging - there's no need to call S.process(), just reset charges to 0
+											S.charge_counter = 0
+				qdel(src)
+				return
+			else
+				to_chat(usr, "<span class='warning'>Only the followers of Nar-Sie may be given their armor.</span>")
+				to_chat(M, "<span class='warning'>Only the followers of Nar-Sie may be given their armor.</span>")
+	to_chat(user, "<span class='note'>You have to be standing on top of the rune.</span>")
+	return
+
+// Animating.
+var/list/word_to_uristrune_table = null
+
+/obj/effect/rune_legacy/proc/get_uristrune_cult(var/word1, var/word2, var/word3, var/mob/M)
+	
+	var/animated
+
+	if((word1 == my_cult.cult_words["travel"] && word2 == my_cult.cult_words["self"])						\
```

Comment:
`Runtime in runes.dm, line 1496: undefined variable /turf/simulated/floor/var/cult_words`